dist/api_data.py

```diff
@@ -1505,6 +1533,10 @@ def __ge__(self, other):
 ]),
 
 'WT_SESSION.flush_tier' : Method([
+    Config('flush_timestamp', '', r'''
+        flush objects to all storage sources using the specified timestamp.
+        The supplied value must not be older than the current oldest timestamp and it must
+        not be newer than the stable timestamp'''),
```

Comment:
FWIW: I think of `flush_tier()` as an operation that is independent of the notions of storage sources and objects. `flush_tier()` simply takes a tiered table and copies keys from the top, writeable tier to the next tier down.  It should work if each tier is a local btree, although that's obviously not the intended use case.

But I'm sure we'll be fine tuning this definition as the system evolves.  So I'm OK leaving this as is for now since it describes what will happen in the less general world we're currently building.

Reply:
I think every tiered table has to be associated with a storage source, even if it is a local source or some kind of no-op source. So I think tiered and storage source are very tied together.

Reply:
As I said, it's not a problem for this PR. But it's something to talk about sometime.